### PR TITLE
feat(math): Support MathML mtext and ms elements

### PR DESCRIPTION
**Pull request by Omikhleia**
Ticking one box in #2138 

I am on a quest, sort of...

So let's derive the [Quadratic Formula](https://developer.mozilla.org/en-US/docs/Web/MathML/Examples/Deriving_the_Quadratic_Formula).

image

### CODE REVIEW

File: packages/math/typesetter.lua [hunk 1]
```diff
@@ -136,6 +136,19 @@ function ConvertMathML (_, content)
       return b.mtr(convertChildren(content))
    elseif content.command == "mtd" then
       return b.stackbox("H", convertChildren(content))
+   elseif content.command == "mtext" or content.command == "ms" then
+      if #content > 1 then
+         SU.error("Wrong number of children in " .. content.command .. ": " .. #content)
+      end
+      local text = content[1] or "" -- empty mtext is allowed, and found in examples...
+      if type(text) ~= "string" then
+         SU.error("ms command contains " .. text .. ", which is not text")
```
Comment (Omikhleia) [inline packages/math/typesetter.lua:145]: Copied for other similar cases from above. But this crashes if text can't be serialized to string.

Comment (Omikhleia) [reply packages/math/typesetter.lua]: Fixed.

**Comment by OlivierNicole** [comment]
This looks like a good working first step. Note that one day we probably want to be able to use SILE commands in `\text`, just like arbitrary TeX formatting can be applied inside a `\text{...}`…

**Comment by Omikhleia** [comment]
> Note that one day we probably want to be able to use SILE commands in `\text`, just like arbitrary TeX formatting can be applied inside a `\text{...}`…

But I'd advocate _against_ it, as far as I have a word to say. It's not a "common" denominator for any real use case.

**Comment by Omikhleia** [comment]
> But I'd advocate _against_ it, as far as I have a word to say. It's not a "common" denominator for any real use case.

"Note that one day we probably want to be able to use SILE commands"... Really? Folks, please be much more precise, do you mean SIL in its TeX-like syntax? (as `\em` and that sort of stuff?). Do you imply there's a real added value _in general_ for this to be true?

Had you said: "one day we probably want to be able to use Markdown, Djot or other markup in `\text`", I'd would have raised an eyebrow with interest.

How long does it take for you to understand that SIL is a dead end?

**Comment by Omikhleia** [comment]
BUT let's go back on topic; there's a PR here, does it pass the bar? Something is better, sometimes, than an idealized nothingness.

**Comment by alerque** [comment]
Hey guys can we take the rhetoric down a notch or two please? I should have caught this earlier but I've spent most of a couple weeks waiting in a hospital and haven't had the energy. This is an entirely volunteer project and interested parties are going to have different ideas and backgrounds. It's okay to disagree, but the last half dozen issues I've looked at today have a bit too much snark. I appreciate the contributions for sure and it's okay to think we're doing some things wrong, but lets take it easy.

Trying to ignore the tone now and address a little point:

> How long does it take for you to understand that SIL is a dead end?

Says who? You don't have to use it or like it, but given how much easier it is to read and write manually than XML I don't see it going anywhere. Also it is (or should be) 100% **just % syntactic sugar over a semantic document** that could just as well have been XML. It is easy to parse and very overhead to maintain, so it is not going anywhere.

That being said I don't see it as any different from any other input format, we should have 100% first class coverage for either XML or SIL (and potentially others in the future), the SIL syntax sugar is not some magic special "this is the SILE way" that gets special treatment.

> "Note that one day we probably want to be able to use SILE commands"... Really? Folks, please be much more precise, do you mean SIL in its TeX-like syntax?

I read this as SILE *commands*, probably matching whatever syntax flavor the parent document is using (XML-ish or TeX-ish or whatever other input module is being used. And yes I agree that text content inside an equation should support the typesetting features of the typesetter, not just plain text. There is precedent for this in every other Math layout system I've seen (including GUI ones used in desktop publishing, for better or worse).

That doesn't mean it has to be in this PR, just saying I'd be open for contributions in that direction.

**Review by alerque — APPROVED** [review]

**Pull request merged**